identity/aziot-identityd/src/lib.rs

```diff
@@ -475,12 +441,32 @@ impl Api {
                 scope_id,
                 attestation,
             } => {
+                async fn operation_to_iot_hub_device(
+                    credentials: aziot_identity_common::Credentials,
+                    operation: aziot_dps_client_async::model::RegistrationOperationStatus,
+                ) -> Result<aziot_identity_common::IoTHubDevice, Error> {
+                    let status = operation.status;
+                    assert!(!status.eq_ignore_ascii_case("assigning"));
```

Comment:
See other comment in the removed `dps_get_registered_device`. This isn't a runtime invariant, it is one of 2-3 possible value that can be returned by DPS

Reply:
Yes, it's not obvious why this would uphold, but it's valid because the underlying `register` call in the DPS client now checks for this case: https://github.com/Azure/iot-identity-service/pull/47/files#diff-50183279457368e456cbe124fbd69c2c4d1d2ebbeb988737299d06b93e78bd4dR151

Reply:
Makes sense in the context of the new code.